staging/src/k8s.io/apimachinery/pkg/runtime/serializer/codec_factory.go

```diff
@@ -227,23 +233,26 @@ func (f CodecFactory) EncoderForVersion(encoder runtime.Encoder, gv runtime.Grou
 	return f.CodecForVersions(encoder, nil, gv, nil)
 }
 
-// DirectCodecFactory provides methods for retrieving "DirectCodec"s, which do not do conversion.
-type DirectCodecFactory struct {
```

Comment:
Can we be less aggressive about removing this? 

This will be a pain for people trying to vendor CRD clients.

For example, this prevents me from updating the kubernetes client prow uses, because prow vendors the knative-build client and that hasn't been updated to require a head version of apimachinery.

```console
vendor/github.com/knative/build/pkg/client/clientset/versioned/typed/build/v1alpha1/build_client.go:81:32: undefined: serializer.DirectCodecFactory
```

Reply:
No objection to a type alias here as well

Reply:
opened https://github.com/kubernetes/kubernetes/pull/76270 to add a type alias to allow existing clients to transition